### DISCUSSION TITLE
Create line type which doesn't overlap with any nodes

### DISCUSSION THREAD

**Discussion by FredrikAugust** [xyflow/xyflow#1577] [New Features]
Do anyone have any guidance on how to create a line which avoids all nodes on the canvas? It would preferably take the shortest non-colliding path from one node to another.

I have an issue where handlers on the right side need to be connected to nodes that are to the left of the node, and right now it looks rather strange as the line goes under the component.
![Screenshot 2021-10-12 at 15 54 22](https://user-images.githubusercontent.com/7691547/136969441-96379b86-a3b3-4ba8-b505-87a4217f349e.png)

**Comment by tisoap** [comment]
I've published a package with a custom edge that does exactly that:
[@tisoap/react-flow-smart-edge](https://www.npmjs.com/package/@tisoap/react-flow-smart-edge)

**Reply by FredrikAugust** [reply]
Was following your PR, and wondered where it went 😅 

Great job, I'll take a look!

**Reply by ottovw** [reply]
Looks great! 

This solution is related to problem / feature request stated here: https://github.com/wbkd/react-flow/discussions/1029
